### PR TITLE
Fix shellcheck warnings and expand travis stylecheck to top-level directory of src

### LINKED ISSUE #44: Adapt tests to work with shellcheck globbing requirements
https://github.com/IBM-Blockchain-Starter-Kit/build-lib/blob/ed4922f57b8f2ef4bbb18e9422884ca141631dbe/src/common/blockchain.sh#L5

When running shellcheck get the following error

```
In src/common/blockchain.sh line 78:
    if ! cf service ${BLOCKCHAIN_SERVICE_INSTANCE} > /dev/null 2>&1
                    ^-- SC2086: Double quote to prevent globbing and word splitting.


In src/common/blockchain.sh line 80:
        cf create-service ${BLOCKCHAIN_SERVICE_NAME} ${BLOCKCHAIN_SERVICE_PLAN} ${BLOCKCHAIN_SERVICE_INSTANCE}
                                                                                ^-- SC2086: Double quote to prevent globbing and word splitting.


In src/common/blockchain.sh line 86:
    if ! cf service-key ${BLOCKCHAIN_SERVICE_INSTANCE} ${BLOCKCHAIN_SERVICE_KEY} > /dev/null 2>&1
                        ^-- SC2086: Double quote to prevent globbing and word splitting.


In src/common/blockchain.sh line 88:
        cf create-service-key ${BLOCKCHAIN_SERVICE_INSTANCE} ${BLOCKCHAIN_SERVICE_KEY}
                              ^-- SC2086: Double quote to prevent globbing and word splitting.


In src/common/blockchain.sh line 94:
    cf service-key ${BLOCKCHAIN_SERVICE_INSTANCE} ${BLOCKCHAIN_SERVICE_KEY} | tail -n +2 > blockchain.json
                   ^-- SC2086: Double quote to prevent globbing and word splitting.


In src/common/blockchain.sh line 160:
    CHAINCODE_FILES=$(find ${CC_PATH} -type f ! -name "*test*")
                           ^-- SC2086: Double quote to prevent globbing and word splitting.


In src/common/blockchain.sh line 299:
                install_fabric_chaincode $CC_ID $CC_VERSION $CC_PATH
                                         ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                            ^-- SC2086: Double quote to prevent globbing and word splitting.


In src/common/blockchain.sh line 311:
                    instantiate_fabric_chaincode $CC_ID $CC_VERSION $channel $CC_INIT_ARGS
                                                 ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                        ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                                    ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                                             ^-- SC2086: Double quote to prevent globbing and word splitting.
```

### PR DESCRIPTION
**Pull request by yorhodes**
Signed-off-by: Yorke Rhodes IV <Yorke.Rhodes.IV@ibm.com>

Closes #44

### CODE REVIEW

File: src/common/blockchain.sh [hunk 1]
```diff
@@ -168,13 +167,14 @@ function install_fabric_chaincode {
...
     
-    CHAINCODE_FILES=$(find ${CC_PATH} -type f ! -name "*test*")
+    CHAINCODE_FILES=$(find "$CC_PATH" -type f ! -name "*test*")
     CHAINCODE_FILE_OPTS=""
     for CHAINCODE_FILE in ${CHAINCODE_FILES}
     do
         CHAINCODE_FILE_OPTS="${CHAINCODE_FILE_OPTS} -F files[]=@${CHAINCODE_FILE}"
     done
 
+    # shellcheck disable=2086
```
Comment (jt-nti) [inline src/common/blockchain.sh:177]: Why is this required?

File: src/router.sh [hunk 2]
```diff
@@ -19,15 +19,16 @@ echo "${stage} stage selected"
...
-    echo ${executable_script}
+    echo "$executable_script"
 else 
     echo "Invalid platform: ${platform} selected"
     exit 1
 fi
 
 # ok now let's call the command we are routing to
-source ${executable_script}
+# shellcheck disable=1090
```
Comment (jt-nti) [inline src/router.sh:33]: Makes sense not to worry about this file, although [the shellcheck doc](https://github.com/koalaman/shellcheck/wiki/SC1090) seems to suggest pointing to `/dev/null` rather than skipping the check...

> If you don't care that ShellCheck is unable to account for the file, specify `# shellcheck source=/dev/null`.

Either way, it would be good to include a comment about why this check can safely be ignored, i.e. the script is intended to select the correct file to source, which is tested elsewhere.

**Review by jt-nti — CHANGES_REQUESTED** [review]
Looks good. Just a couple of minor comments.

File: src/common/blockchain.sh (hunk 1 above)
Comment (yorhodes) [reply src/common/blockchain.sh:177]: The way the CHAINCODE_FILE_OPTS variable is built, in order for it to work within the curl call, it cannot be wrapped in quotes (which shellcheck requires around variables). If you can find a way around this, I will gladly accept it. This was the lazy fix to get the rest of the file compliant at least.

File: src/router.sh (hunk 2 above)
Comment (yorhodes) [reply src/router.sh:33]: Hmm, yes redirecting to null seems to be the SC1090 specific way to fix. I was using more generic.

File: src/common/blockchain.sh (hunk 1 above)
Comment (jt-nti) [reply src/common/blockchain.sh:177]: Ok, I think I see what the problem is now. I think there may be a better option than multiple `-F` arguments- I'll open an issue to have a look at some point.

**Comment by jt-nti** [comment]
I've created issue #67 so could you rebase this so I can approve and merge it please @yorhodes 
Rebased here if that helps -> https://github.com/jt-nti/build-lib/tree/globfix-rebase
Thanks

**Comment by jt-nti** [comment]
Delivered in #73
